roiextractors/extractors/nwbextractors/nwbextractors.py

```diff
@@ -350,6 +342,40 @@ def add_epochs(imaging, nwbfile):
 
         return nwbfile
 
+    @staticmethod
+    def get_nwb_metadata(imgextractor: ImagingExtractor):
+        """
+        Converts metadata from the segmentation into nwb specific metadata
+        Parameters
+        ----------
+        imgextractor: ImagingExtractor
+        """
+        metadata = get_default_nwb_metadata()
+        # Optical Channel name:
+        for i in range(imgextractor.get_num_channels()):
+            ch_name = imgextractor.get_channel_names()[i]
+            if i == 0:
+                metadata['Ophys']['ImagingPlane'][0]['optical_channel'][i]['name'] = ch_name
+            else:
+                metadata['Ophys']['ImagingPlane'][0]['optical_channel'].append(dict(
+                    name=ch_name,
+                    emission_lambda=np.nan,
+                    description=f'{ch_name} description'
+                ))
+
+        # set imaging plane rate:
+        rate = np.float(
+            'NaN') if imgextractor.get_sampling_frequency() is None else imgextractor.get_sampling_frequency()
+        # adding imaging_rate:
+        metadata['Ophys']['ImagingPlane'][0].update(imaging_rate=rate)
+        # TwoPhotonSeries update:
+        metadata['Ophys']['TwoPhotonSeries'][0].update(
+            dimension=imgextractor.get_image_size())
+        # remove what Segmentation extractor will input:
+        _ = metadata['Ophys'].pop('ImageSegmentation')
+        _ = metadata['Ophys'].pop('Fluorescence')
```

Comment:
Seems odd to add these as default and then remove them. If you only want them for Segmentation extractor, then add them there

Reply:
there are fewer lines of code and I feel its more readable to remove unneeded fields than add the whole dict to a field (fewer lines)